### ISSUE TITLE
suboptimal GPU utilization for non-uniform quantization if nql_opt_mode=both

### ISSUE THREAD

**Issue by rapwag01** [Tencent/PocketFlow#273]
**Describe the bug**
When updating cluster points as well as weights during nonuniform quantization
(--nuql_opt_mode=both), almost half of a training iteration gets executed on CPU - even though a GPU is available.

I've dumped some run meta data and inspected it using Chrome trace (screenshot). You can see that the "Unique" and "UnsortedSegmentSum" operations run on CPU rather than GPU.

Op names:

Adam/update_model/MobilenetV1/Logits/Conv2d_1c_1x1/Conv2D/nonuniform_quantize/clusters/Unique

Adam/update_model/MobilenetV1/Logits/Conv2d_1c_1x1/Conv2D/nonuniform_quantize/clusters/UnsortedSegmentSum

When using --nuql_opt_mode=weights, the two operations disappear and GPU utilization is significantly better. 

Can you help me figuring out how to improve the cluster-optimization step to be run on GPU? I cannot find the two ops mentioned above in Tensorboard. Where in the source code (learners/nonuniform_quantization/* files) do you update the cluster-point positions using AdamOptimizer? 

**To Reproduce**
1) Implement TF profiler as explained here https://towardsdatascience.com/howto-profile-tensorflow-1a49fb18073d
2) Run non-uniform quantization learner on mobilenetv1 trained on imagenet with default settings
3) Inspect trace-files in Chrome (see point 1))

**Expected behavior**
Run cluster update on GPU. 

**Screenshots**
![image](https://user-images.githubusercontent.com/46925691/55490867-66843180-562c-11e9-9aaa-8b1ff8e1064c.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 14.02
